src/services/localStorage.ts

```diff
@@ -12,3 +14,9 @@ export function getFromLocalStorage(key: string): string {
   }
   return data;
 }
+
+export const getFavoriteCities = (): LocationInfoType[] => {
+  const favoriteCities: LocationInfoType[] = (getFromLocalStorage('favorite-cities') ||
```

Comment:
'favorite-cities' string can be defined as separate exported const

Reply:
can you please share code snippet of what you've done? using variable instead of hardcoded string should not work like you're describing

Reply:
In localStorage.ts
`export const favoriteCitiesConst: LocationInfoType[] = (getFromLocalStorage('favorite-cities') ||
    []) as LocationInfoType[]`

In MainCard.tsx
`const toggleFavoriteCities = () => {
    let favoriteCities = favoriteCitiesConst //there
    console.log(favoriteCities)
    if (!isFavorite) {
      favoriteCities = favoriteCities.concat(location);
    } else {
      favoriteCities = favoriteCities.filter(city => city.id !== location.id);
    }
    setInLocalStorage(favoriteCities, 'favorite-cities');
  };`

for all click in btn I have same favoriteCities.

Reply:
I don't know why I didn't understand you right away 😅
Sorry:)